src/pseudopeople/configuration/validator.py

```diff
@@ -93,105 +101,90 @@ def _get_default_config_node(
-                f"column {column}."
-            )
-
+            raise ConfigurationError(invalid_error + f"'{parameter}' cannot include 0.")
     if isinstance(noise_type_config, Dict):
         for value in noise_type_config.values():
-            if not isinstance(value, float):
-                raise TypeError(
-                    "All possible age miswriting probabilities must be floats. "
-                    f"Provided {value} of type {type(value)} in the configuration "
-                    f"for form {form} and column {column}."
+            if not isinstance(value, (float, int)):
+                raise ConfigurationError(
+                    invalid_error + f"'{parameter}' probabilities must be floats or ints. "
+                    f"Provided {value} of type {type(value)}."
+                )
+            if not (0 <= value <= 1):
+                raise ConfigurationError(
+                    invalid_error
+                    + f"'{parameter}' probabilities must be between 0 and 1 (inclusive). "
+                    f"Provided {value} in {list(noise_type_config.values())}."
                 )
-
         if not np.isclose(sum(noise_type_config.values()), 1.0):
-            raise ValueError(
-                "The provided age miswriting probabilities must sum to 1. "
+            raise ConfigurationError(
+                invalid_error + f"'{parameter}' probabilities must sum to 1. "
                 f"Provided values sum to {sum(noise_type_config.values())}."
             )
 
 
 def _validate_zipcode_digit_probabilities(
-    noise_type_config: List, form: str, column: str, parameter: str, noise_type: str
+    noise_type_config: List, parameter: str, invalid_error: str
 ) -> None:
     """Validates the user-provided values for the zipcode digit noising probabilities"""
     if not isinstance(noise_type_config, List):
-        raise TypeError(
-            f"Invalid configuration type provided for '{parameter}'; must be a list. "
-            f"Provided {noise_type_config} of type {type(noise_type_config)} in the "
-            f"configuration for noise type {noise_type} in form {form} and column {column}."
+        raise ConfigurationError(
+            invalid_error + f"'{parameter}' must be a List. "
+            f"Provided {noise_type_config} of type {type(noise_type_config)}."
+        )
+    if len(noise_type_config) != 5:
+        raise ConfigurationError(
+            invalid_error + f"'{parameter}' must be a List of 5 probabilities. "
+            f"{len(noise_type_config)} probabilities ({noise_type_config})."
         )
-
     for value in noise_type_config:
-        if not isinstance(value, float):
-            raise TypeError(
-                "All possible zipcode digit miswriting probabilities must be floats. "
-                f"Provided {value} of type {type(value)} in the configuration "
-                f"for form {form} and column {column}."
+        if not isinstance(value, (float, int)):
```

Comment:
I believe this is duplicated code. You can just call `_validate_default_standard_parameters` inside this loop?

Reply:
Nice, good catch